### DISCUSSION TITLE
Change the Default_root_dir in YAML File

### DISCUSSION THREAD

**Discussion by yxteo2** [open-edge-platform/anomalib#2477] [Q&A]
Hi guys, how can I change the root directory in YAML file.
I try to:

`datamodule.trainer.default_root_dir = THE DIRECTORY I WANT`

but I got this

`╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ in <module>:1                                                                                    │
│                                                                                                  │
│ ❱ 1 datamodule.trainer.default_root_dir                                                          │
│   2                                                                                              │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'NoneType' object has no attribute 'default_root_dir'`

**Comment by blaz-r** [comment]
I'm not sure what exactly you want to achieve.

If you want to change the location of **data**, you'll need to use `root` inside datamodule init args:
```yaml
class_path: anomalib.data.MVTec
init_args:
  root: ../datasets/MVTec
```
If you want to change the `default_root_dir` of trainer (where the results are stored etc.), then you must use the following lines:
```yaml
trainer:
   default_root_dir: <insert dirname>
```

(don't nest it inside `datamodule` key).

**Reply by yxteo2** [reply]
I am trying to change the root directory after loading the yaml file. 
But with 
`Engine.from_config(CONFIG_PATH)`
I cant adjust the root directory

**Reply by alexriedel1** [reply]
then you have to monkey patch it like:
```
engine = Engine.from_config(CONFIG_PATH)
engine.default_root_dir = dir
```

but the better way would be to set the root dir in your config..
